src/pages/index.js

```diff
@@ -166,11 +205,76 @@ function getCardElement(data) {
 const cardsList = document.querySelector(".cards__list");
 
 // Loop through the initialCards array and append each card to the parent
+/*
 initialCards.forEach((cardData) => {
   renderCard(cardData, "append");
 });
+*/
 
 function renderCard(item, method = "prepend") {
   const cardElement = getCardElement(item);
   cardsList[method](cardElement);
 }
+
+
+// functions
+console.log(deleteModal)
+deleteModalDeleteBtn.addEventListener('click', () => {
+  //console.log(currentCardToDelete)
+  deleteModalSubmitButton.textContent = 'Deleting...'
+  api.deleteCard(selectedCardId).then((dd) => {
+    selectedCardEl.remove()
+    closeModal(deleteModal)
+  }).catch((err) => console.error(err)).finally(() => deleteModalSubmitButton.textContent = 'Delete')
+})
+
+deleteModalCancleBtn.addEventListener('click', () => {
+  closeModal(deleteModal)
+})
+
+// user information
+function displayUserInformation() {
+  api.getUserInformation().then((dd) => {
+    profileNameElement.textContent = dd.name
+    profileJobElement.textContent = dd.about
+    profileAvatartElement.setAttribute('src', dd.avatar)
+    console.log(dd)
+  }).catch((err) => console.error(err))
+}
+
+// initial cards
+function displayInitialCards() {
+  api.getInitialCards().then((dd) => {
+    console.log(dd)
+    dd.forEach((cardData) => {
+      renderCard(cardData, "append");
+    });
+    }).catch((err) => console.error(err))
+}
+
+function handleAddCardSubmit(e) {
+  e.preventDefault();
+  cardModalSubmitButton.textContent = 'Saving...'
+  const data = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  }
+  api.addNewCard(data).then((dd) => {
+      //console.log(dd)
+      renderCard(dd)
+}).catch((err) => console.error(err)).finally(() => cardModalSubmitButton.textContent = 'Save');
+  e.target.reset();
+  //disable submit button
+  disableButton(cardModalSubmitButton);
+  closeModal(cardModal);
```

Comment:
```js

  e.target.reset();
  //disable submit button
  disableButton(cardModalSubmitButton);
  closeModal(cardModal);
```

These 3 lines of code should be in `.then` in line 263


```js
  api
    .addNewCard(data)
    .then((dd) => {
      e.target.reset();
      disableButton(cardModalSubmitButton);
      closeModal(cardModal);
      renderCard(dd);
    })
    .catch((err) => console.error(err))
    .finally(() => (cardModalSubmitButton.textContent = "Save"));
}

```

Reply:
You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.